metadata-ingestion/src/datahub/ingestion/source/state/checkpoint.py

```diff
@@ -50,7 +49,13 @@ def to_bytes(
         if self.serde == "utf-8":
             encoded_bytes = CheckpointStateBase._to_bytes_utf8(self)
         elif self.serde == "base85":
-            encoded_bytes = CheckpointStateBase._to_bytes_base85(self, compressor)
+            # The original base85 implementation used pickle, which would cause
+            # issues with deserialization if we ever changed the state class definition.
+            raise ValueError(
```

Comment:
why not default to using base85-bz2-json when base85 is specified? That way production ingestion pipelines can continue to work?

Reply:
I see you are doing basically that during deserialization anyway!

Reply:
when will this code path be exercised?